### ISSUE TITLE
Content Too Big

### ISSUE THREAD

**Issue by bodysoda69** [AnimeKaizoku/ArchivsitsBot#8]
File "/usr/local/lib/python3.9/site-packages/telegraph/api.py", line 164, in create_page
    return self._telegraph.method('createPage', values={
  File "/usr/local/lib/python3.9/site-packages/telegraph/api.py", line 37, in method
    raise TelegraphException(response.get('error'))
telegraph.exceptions.TelegraphException: CONTENT_TOO_BIG

Any solution for this?

**Comment by HikaruHokkyokusei** [comment]
I think telegraph has a limit of maximum content that can be written in a page.

In our code, we create the telegraph page in the bot/helper/drive_utils/gdriveTools.py

We write some content to the page for each url that we mention. So, we have put a limit of only including maximum of 95 urls.

Now, I am not exactly sure where the limits is for the content, or for the number of urls that we include, but there is some limit which is either of the two.

**Comment by bodysoda69** [comment]
> I think telegraph has a limit of maximum content that can be written in a page.
> 
> In our code, we create the telegraph page in the bot/helper/drive_utils/gdriveTools.py
> 
> We write some content to the page for each url that we mention. So, we have put a limit of only including maximum of 95 urls.
> 
> Now, I am not exactly sure where the limits is for the content, or for the number of urls that we include, but there is some limit which is either of the two.

What's the use of Previous and Next buttons? I suppose it's there to create a new page in telegraph when limit exceeds right? 

But the main issue is the bot gets stuck in Searching, if it gets hit by this error is it possible to send message like content too big, refine keyword. That'd be nice too.

**Comment by HikaruHokkyokusei** [comment]
Hmm, Yea, I have not actually done anything on the telegraph part on this bot, as it was built by previous creator, so, I am not sure how telegraph works.

But If Next and previous button exists, then yea, i think they are for the time when content limit is exceeded. I'll check the docs and see how to use these next and previous buttons.

As for the fact that the bot is getting stuck in searching, We have the 95 limit to not exceed the content, but there are times, when the bot says "Searching..." and it never updates the message. So, I think the cause is this telegraph limit. I'll add try / catch and then send a msg "Refine Search keyword" for the time being. 

Then, I'll see whete I can use those next page system so as to include more results.

**Comment by bodysoda69** [comment]
> As for the fact that the bot is getting stuck in searching, We have the 95 limit to not exceed the content, but there are times, when the bot says "Searching..." and it never updates the message. So, I think the cause is this telegraph limit. I'll add try / catch and then send a msg "Refine Search keyword" for the time being.

This would be very helpful for now, Thanks!

**Comment by ghost** [comment]
> Hmm, Yea, I have not actually done anything on the telegraph part on this bot, as it was built by previous creator, so, I am not sure how telegraph works.
> 
> But If Next and previous button exists, then yea, i think they are for the time when content limit is exceeded. I'll check the docs and see how to use these next and previous buttons.
> 
> As for the fact that the bot is getting stuck in searching, We have the 95 limit to not exceed the content, but there are times, when the bot says "Searching..." and it never updates the message. So, I think the cause is this telegraph limit. I'll add try / catch and then send a msg "Refine Search keyword" for the time being.
> 
> Then, I'll see whete I can use those next page system so as to include more results.

Yeah doing this will be helpful. Something like when telegraph limit exceeds shift to next page else if it's under Telegraph limit then show results

**Comment by HikaruHokkyokusei** [comment]
In the docs, they have said that telegraph page size can only be upto 64 kB. What I will do is, add a next and prev button at the bottom of the html page and put anchon tags on them. When ever a limit is hit, a new page will be created and thier link will be added there in those next and prev buttons.

**Comment by bodysoda69** [comment]
Damn, You're fast. Waiting!

**Comment by HikaruHokkyokusei** [comment]
Ok, So in the last commint, I added the try catch in case the content size exceeds 64 kB during page creation, and reverted the Next page and prev page system. The old code in the file `bot/helper/drive_utils/gdriveTools.py` already contained a function (called "edit_telegraph") that can append "Next" and "Previous" button at the bottom of the page, but the old code never used that function, and I will continue to do the same. The reason beign I'll have to add a blocking code to wait few milliseconds between page creations back to 1 as telgraph has a flood limiter. So, to go through all the hassel, I just reverted it to good old technique by limiting results to 95 max. In old code, I tried this without knowing about the flood restrictions and then we had to change the api key.

With this, I'll close this issue.
